fvm/environment/event_emitter.go

```diff
@@ -288,7 +288,6 @@ func (collection *EventCollection) AppendServiceEvent(
 	collection.convertedServiceEvents = append(
 		collection.convertedServiceEvents,
 		*convertedEvent)
-	collection.eventCounter++
```

Comment:
We need to fix this test case, right?
https://github.com/onflow/flow-go/pull/3386/files#diff-5abed056e7e55b2bbe2a0066aa4de2b3cbb8964ba055f805d6728bb7c78401edR28

Reply:
yea, I'm waiting for us to decide on the right approach before updating tests

Reply:
OK, the fix looks good to me.

Reply:
Looking at the (usage)[https://github.com/onflow/flow-go/blob/master/fvm/environment/event_emitter.go#L167] and the fix looks good to me too.